### PR TITLE
copr: add instr

### PR DESCRIPTION
**Pull request by wangggong**
Signed-off-by: wangggong <793160615@qq.com>

### What problem does this PR solve?

Issue Number: #9016 

Problem Summary:

### What is changed and how it works?

What's Changed:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

### Release note 

No release note.

### CODE REVIEW

File: components/tidb_query_expr/src/impl_string.rs [hunk 1]
```diff
@@ -3218,6 +3226,54 @@ mod tests {
...
+            (Some(b" ".to_vec()), Some(b"abcdefg".to_vec()), Some(0)),
+            (Some(b"".to_vec()), Some(b"".to_vec()), Some(1)),
+            (Some(b" ".to_vec()), Some(b"".to_vec()), Some(0)),
+            (Some(b"".to_vec()), Some(b" ".to_vec()), Some(1)),
+            (Some(b"eFg".to_vec()), Some(b"abcdefg".to_vec()), Some(0)),
+            (Some(b"deF".to_vec()), Some(b"abcdefg".to_vec()), Some(0)),
+            (
+                Some("字节".as_bytes().to_vec()),
+                Some("a多字节".as_bytes().to_vec()),
+                Some(1 + "a多字节".find("字节").unwrap() as i64),
```
Comment (Xuanwo) [inline components/tidb_query_expr/src/impl_string.rs:3246]: Please give the actual answer instead fo a call to `find`.

Comment (wangggong) [reply components/tidb_query_expr/src/impl_string.rs]: done

PTAL

**Review by Xuanwo — APPROVED** [review]

**Review by skyzh — APPROVED** [review]
LGTM

**Comment by skyzh** [comment]
By the way, I think implementation of `instr_utf8` is wrong. It always convert string to lowercase. If you are interested in that, you may also help propose a fix. FYI, https://github.com/pingcap/tidb/blob/master/expression/builtin_string.go#L3758

**Comment by hi-rustin** [comment]
> /merge

You only need to trigger /merge once, and if the CI test fails, you just re-trigger the test that failed and the bot will merge the PR for you after the CI passes.

**Pull request merged**